fulcio-spec.md

```diff
@@ -0,0 +1,307 @@
+
+See [Spec: Fulcio](https://docs.google.com/document/d/1W5xp3g8_jaqzDQmIvepNYsWb-bQNc0U2ZQgQ700Kjok/edit#heading=h.lrl46qy6806e) for more information about required certificate extensions.
+
+### 2.3. Keys
+
+Note that Fulcio can be used with self-managed keys, as described in a [Sigstore blog post](https://blog.sigstore.dev/adopting-sigstore-incrementally-1b56a69b8c15). Fulcio does not enforce the use of ephemeral keys. The signer may have experience with managing keys and appropriately locking down the signing key, for example through the use of an HSM or ACLs. Fulcio still provides benefits through a tighter integration with Sigstore and simplifying the verification policy.
+
+## 3. Issuance - Life of a Request
+
+The client submits a certificate request to Fulcio. The certificate request MUST contain a certificate request described subsequently and OpenID Connect (OIDC) identity token. This is a signed JWT containing information about the principal (identity of the client), the issuer (who issued the identity token - Google, Microsoft, GitHub, etc.) and additional metadata such as expiration. The principal identity can either be a signing identity in the form of an email or username, or a workload identity. The certificate request MUST contain either:
+
+- A public key and signed challenge. This is the public portion of a cryptographic key pair generated by the client. The public key will be embedded in the issued X.509 certificate. The challenge proves the client is in possession of the private key that corresponds to the public key provided. The challenge SHOULD be created by signing the subject (`sub`) of the OIDC identity token.
+- A PKCS#10 ([RFC2986](https://www.rfc-editor.org/rfc/rfc2986)) certificate signing request (CSR), which also provides a proof of possession and the public key. The CSR subject MAY contain the subject of the OIDC ID token, but there is no mandate to do so, as Fulcio will not check that the subject of the CSR matches the subject of the token.
+
+Fulcio MUST authenticate the OIDC ID token. To authenticate, Fulcio MUST follow the [OIDC specification](https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation), which includes the following steps:
+
+- Use the issuer claim from the token to find the issuer's OIDC discovery endpoint
+- Download the issuer's verification keys from the discovery endpoint
+- Verify the ID token signature
+
+Fulcio does not support MAC-based authentication.
+
+Once the client has been authenticated, Fulcio MUST verify the client is in possession of the private key of the public key they’ve submitted. Fulcio MUST verify the signed challenge or CSR. For a signed challenge, this MUST be a signature over the identity claim of the ID token, which SHOULD be the `sub` claim but MAY be a non-standard claim as Fulcio supports configuration of this claim. The challenge and CSR are verified using the provided public key.
```

Comment:
Is 'token' meant instead of 'client' in 'Once the client has been authenticated'?

Reply:
Yes, thanks, `token` would be more accurate than `client` - "Once the token has been authenticated".